### PR TITLE
fix: generate view ids

### PR DESCRIPTION
**Pull request by uziasferreirazup**
### Related Issues

close issue #742

### Description and Example
Create logic to generate view ids and reuse these ids when the app rotate

### Checklist

Please, check if these important points are met using `[x]`:

- [X] I read the [PR Guide] and followed the process outlined there for submitting this PR.
- [X] I avoided _breaking changes_ by not changing public APIs that people rely on. 
- [X] I am willing to follow-up on review comments in a timely manner.

[PR Guide]: https://github.com/ZupIT/beagle/blob/master/doc/contributing/pull_requests.md

### CODE REVIEW

**Comment by botbeaglezup** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Big PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: android/beagle/src/main/java/br/com/zup/beagle/android/view/viewmodel/GenerateIdViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+    fun createIfNotExisting(parentId: Int) {
+        val view = views[parentId]
+        if (view == null) {
+            views[parentId] = LocalView()
+        }
+    }
+
+    fun getViewId(parentId: Int): Int {
+        val view = views[parentId]!!
```
[the remark this answers is not in the document]
Comment (uziasferreirazup) [reply android/beagle/src/main/java/br/com/zup/beagle/android/view/viewmodel/GenerateIdViewModel.kt]: Done..

File: android/beagle/src/main/java/br/com/zup/beagle/android/view/viewmodel/GenerateIdViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+    fun createIfNotExisting(parentId: Int) {
+        val view = views[parentId]
+        if (view == null) {
+            views[parentId] = LocalView()
+        }
+    }
+
+    fun getViewId(parentId: Int): Int {
+        val view = views[parentId] ?: throw Exception(PARENT_ID_NOT_FOUND)
```
Comment (botbeaglezup) [inline android/beagle/src/main/java/br/com/zup/beagle/android/view/viewmodel/GenerateIdViewModel.kt:38]: <table data-meta="generated_by_android">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/8481995986323a9c27eadb36b44c257ee8e1245b/android/beagle/src/main/java/br/com/zup/beagle/android/view/viewmodel/GenerateIdViewModel.kt#L38"></span>Exception is a too generic Exception. Prefer throwing specific exceptions that indicate a specific error case.</td>
    </tr>
  </tbody>
</table>

File: android/beagle/src/main/java/br/com/zup/beagle/android/view/viewmodel/GenerateIdViewModel.kt [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        return if (!view.created) generateNewViewId(view) else view.temporaryIds.pollFirst()
+            ?: throw Exception("temporary ids can't be empty")
```
Comment (botbeaglezup) [inline android/beagle/src/main/java/br/com/zup/beagle/android/view/viewmodel/GenerateIdViewModel.kt:40]: <table data-meta="generated_by_android">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/8481995986323a9c27eadb36b44c257ee8e1245b/android/beagle/src/main/java/br/com/zup/beagle/android/view/viewmodel/GenerateIdViewModel.kt#L40"></span>Exception is a too generic Exception. Prefer throwing specific exceptions that indicate a specific error case.</td>
    </tr>
  </tbody>
</table>

File: android/beagle/src/main/java/br/com/zup/beagle/android/view/viewmodel/GenerateIdViewModel.kt [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    }
+
+    private fun generateNewViewId(view: LocalView): Int {
+        val id = View.generateViewId()
+        view.ids.add(id)
+        return id
+    }
+
+    fun setViewCreated(parentId: Int) {
+        if (views[parentId] == null) throw Exception(PARENT_ID_NOT_FOUND)
```
Comment (botbeaglezup) [inline android/beagle/src/main/java/br/com/zup/beagle/android/view/viewmodel/GenerateIdViewModel.kt:50]: <table data-meta="generated_by_android">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/8481995986323a9c27eadb36b44c257ee8e1245b/android/beagle/src/main/java/br/com/zup/beagle/android/view/viewmodel/GenerateIdViewModel.kt#L50"></span>Exception is a too generic Exception. Prefer throwing specific exceptions that indicate a specific error case.</td>
    </tr>
  </tbody>
</table>

File: android/beagle/src/main/java/br/com/zup/beagle/android/view/viewmodel/GenerateIdViewModel.kt [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+    private val views = mutableMapOf<Int, LocalView>()
+
+    fun createIfNotExisting(parentId: Int) {
+        val view = views[parentId]
+        if (view == null) {
+            views[parentId] = LocalView()
+        }
+    }
+
+    fun getViewId(parentId: Int): Int {
```
Comment (botbeaglezup) [inline android/beagle/src/main/java/br/com/zup/beagle/android/view/viewmodel/GenerateIdViewModel.kt:38]: <table data-meta="generated_by_android">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/12520afe7dec8d2710e889e87ebfd884dcf37a22/android/beagle/src/main/java/br/com/zup/beagle/android/view/viewmodel/GenerateIdViewModel.kt#L38"></span>Exception is a too generic Exception. Prefer throwing specific exceptions that indicate a specific error case.</td>
    </tr>
  </tbody>
</table>

File: android/beagle/src/main/java/br/com/zup/beagle/android/view/viewmodel/GenerateIdViewModel.kt [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+        val view = views[parentId] ?: throw BeagleException(PARENT_ID_NOT_FOUND)
+        return if (!view.created) generateNewViewId(view) else view.temporaryIds.pollFirst()
```
Comment (botbeaglezup) [inline android/beagle/src/main/java/br/com/zup/beagle/android/view/viewmodel/GenerateIdViewModel.kt:40]: <table data-meta="generated_by_android">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/12520afe7dec8d2710e889e87ebfd884dcf37a22/android/beagle/src/main/java/br/com/zup/beagle/android/view/viewmodel/GenerateIdViewModel.kt#L40"></span>Exception is a too generic Exception. Prefer throwing specific exceptions that indicate a specific error case.</td>
    </tr>
  </tbody>
</table>

File: android/beagle/src/main/java/br/com/zup/beagle/android/view/viewmodel/GenerateIdViewModel.kt [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+            ?: throw BeagleException("temporary ids can't be empty")
+    }
+
+    private fun generateNewViewId(view: LocalView): Int {
+        val id = View.generateViewId()
+        view.ids.add(id)
+        return id
+    }
+
+    fun setViewCreated(parentId: Int) {
```
Comment (botbeaglezup) [inline android/beagle/src/main/java/br/com/zup/beagle/android/view/viewmodel/GenerateIdViewModel.kt:50]: <table data-meta="generated_by_android">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/12520afe7dec8d2710e889e87ebfd884dcf37a22/android/beagle/src/main/java/br/com/zup/beagle/android/view/viewmodel/GenerateIdViewModel.kt#L50"></span>Exception is a too generic Exception. Prefer throwing specific exceptions that indicate a specific error case.</td>
    </tr>
  </tbody>
</table>

File: android/beagle/src/main/java/br/com/zup/beagle/android/view/viewmodel/GenerateIdViewModel.kt [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+        if (views[parentId] == null) throw BeagleException(PARENT_ID_NOT_FOUND)
+
+        views[parentId] = views[parentId]!!.apply {
```
Comment (matheusribeirozup) [inline android/beagle/src/main/java/br/com/zup/beagle/android/view/viewmodel/GenerateIdViewModel.kt:53]: The same here

Comment (uziasferreirazup) [reply android/beagle/src/main/java/br/com/zup/beagle/android/view/viewmodel/GenerateIdViewModel.kt:53]: check the line 51

**Pull request merged**